src/common/io.py

```diff
@@ -2990,6 +2990,163 @@ def get_result(self) -> ResultDymolaBinary:
+
+        data_sections = ["name", "dataInfo", "data_2", "data_3"]
+        with open(self._fname, "rb") as f:
+            delayed = _DelayedVariableLoadDiags(f, chars_as_strings=False)
+            self.raw: dict = delayed.get_variables(variable_names = data_sections)
+
+        self._name_info: dict = self.raw["name"]
+        self._dataInfo: dict = self.raw["dataInfo"]
+        self._data_2_info: dict = self.raw["data_2"]
+        self._contains_diagnostic_data: bool = ("data_3" in self.raw)
+        if self._contains_diagnostic_data:
+            self._data_3_info: dict = self.raw["data_3"]
+
+    @cache
+    def _get_variable_name_to_index_dict(self) -> dict[str, int]:
+        name_dict: dict = fmi_util.read_name_list(
+            encode(self._fname),
+            self._name_info["file_position"],
+            int(self._name_info["nbr_variables"]),
+            int(self._name_info["max_length"])
+        )
+
+        return {decode(k): v for k, v in name_dict.items()}
+
+    @cache
+    def get_variable_names(self) -> list[str]:
+        return list(self._get_variable_name_to_index_dict().keys())
+
+    def _get_data_index_mat(self, variable_name: str) -> tuple[int, int]:
+        """ Returns the data index and matrix ID for given variable name. """
+        if variable_name in self._get_variable_name_to_index_dict():
+            variable_index = self._get_variable_name_to_index_dict().get(variable_name)
+            data_mat = self._dataInfo[0][variable_index]
+            data_index = abs(self._dataInfo[1][variable_index]) - 1 # abs due to alias
+
+            if data_mat == 0:
+                data_mat = 2 if len(self.raw['data_2']) > 0 else 1
+
+            return data_index, data_mat
+
+        raise KeyError("Could not find variable")
+
+    @cached_property
+    def _data_1(self):
+        """Non time-varying values, i.e., constants and fixed parameters."""
+        return scipy.io.loadmat(self._fname, chars_as_strings=False, variable_names=["data_1"])["data_1"]
+
+    def get_trajectory(self, name: str) -> Trajectory:
+        time = self._get_diagnostics_trajectory(0) if self._contains_diagnostic_data else self._get_trajectory(0)
+
+        if name in ('time', 'Time'):
+            return Trajectory(time, time)
+
+        data_index, data_mat = self._get_data_index_mat(name)
+        match data_mat:
+            case 1:
+                return Trajectory(self._data_1[0], self._data_1[data_index])
+            case 2:
+                data = (
+                    self._get_interpolated_trajectory(data_index)
+                    if self._contains_diagnostic_data
+                    else self._get_trajectory(data_index)
+                )
+                return Trajectory(time, data)
+            case 3:
+                return Trajectory(time, self._get_diagnostics_trajectory(data_index))
+            case _:
+                raise ValueError(f"Invalid data matrix: {data_mat}")
+
+    @cache
```

Comment:
Why does all the get_trajectory functions have caching? I would prefer if they did not have caching as to keep things more flexible, caching for trajectories should probably not be at this level. I guess we might want caching for the 'time' variable as it is used many times internally. Maybe we can create separate methods for getting the time variables and add cache on them?

Reply:
Note: We have this type of caching in `ResultDymolaBinary`, but it does not make too much sense on reflection, except for time.